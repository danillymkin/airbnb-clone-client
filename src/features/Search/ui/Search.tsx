import React, { FunctionComponent } from 'react';
import { BiSearch } from 'react-icons/bi';

export const Search: FunctionComponent = (): JSX.Element => {
  return (
    <div
      className={
        'w-full cursor-pointer rounded-full border py-2 shadow-sm transition hover:shadow-md md:w-auto'
      }
      role={'search'}
    >
      <div className={'flex items-center justify-between'}>
        <div className={'px-6 text-sm font-semibold'}>Anywhere</div>
        <div
          className={
            'hidden flex-1 border-x px-6 text-center text-sm font-semibold sm:block'
          }
        >
          Any Week
        </div>
        <div
          className={'flex items-center gap-3 pl-6 pr-2 text-sm text-gray-600'}
        >
          <div className={'hidden sm:block'}>Add Guests</div>
          <div className={'rounded-full bg-rose-500 p-2 text-white'}>
            <BiSearch role={'img'} />
          </div>
        </div>
      </div>
    </div>
  );
};
