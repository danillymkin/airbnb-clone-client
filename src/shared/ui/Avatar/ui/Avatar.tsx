import React, { FunctionComponent } from 'react';
import Image from 'next/image';

interface AvatarProps {}

type Props = AvatarProps;

export const Avatar: FunctionComponent<Props> = ({}): JSX.Element => {
  return (
    <Image
      className={'rounded-full'}
      src={'/images/avatar-fallback.jpg'}
      width={30}
      height={30}
      alt={'Avatar'}
    />
  );
};
