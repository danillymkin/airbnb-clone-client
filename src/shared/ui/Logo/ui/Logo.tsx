'use client';

import React, { FunctionComponent } from 'react';
import Image from 'next/image';

interface LogoProps {}

type Props = LogoProps;

export const Logo: FunctionComponent<Props> = ({}): JSX.Element => {
  return (
    <Image
      alt={'Logo'}
      className={'hidden cursor-pointer md:block'}
      height={31}
      width={100}
      src={'/images/logo.png'}
      priority
    />
  );
};
