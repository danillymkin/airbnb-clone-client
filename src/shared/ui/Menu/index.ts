export { Menu } from './ui/Menu/Menu';
export { MenuItem } from './ui/MenuItem/MenuItem';
