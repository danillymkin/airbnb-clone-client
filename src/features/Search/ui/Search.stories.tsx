import { Meta, StoryObj } from '@storybook/react';
import { Search } from './Search';

const meta: Meta = {
  title: 'Features/Search',
  component: Search,
};

export default meta;

type Story = StoryObj<typeof Search>;

export const Default: Story = {
  render: () => <Search />,
};
