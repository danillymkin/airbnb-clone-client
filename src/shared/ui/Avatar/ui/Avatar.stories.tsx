import { Meta, StoryObj } from '@storybook/react';
import { Avatar } from '@/shared/ui/Avatar';

const meta: Meta = {
  title: 'UI/Avatar',
  component: Avatar,
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  render: () => <Avatar />,
};
