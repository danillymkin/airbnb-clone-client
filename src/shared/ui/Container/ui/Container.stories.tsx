import { Meta, StoryObj } from '@storybook/react';
import { Container } from '@/shared/ui/Container';

const meta: Meta = {
  title: 'UI/Container',
  component: Container,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof Container>;

export const Default: Story = {
  render: () => (
    <Container>
      <div>Children</div>
    </Container>
  ),
};
