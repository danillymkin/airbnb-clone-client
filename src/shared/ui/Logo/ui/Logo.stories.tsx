import { Meta, StoryObj } from '@storybook/react';
import { Logo } from '@/shared/ui/Logo';

const meta: Meta<typeof Logo> = {
  title: 'UI/Logo',
  component: Logo,
};

export default meta;

type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  render: () => <Logo />,
};
