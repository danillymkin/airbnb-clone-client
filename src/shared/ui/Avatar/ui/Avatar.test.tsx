import { render, screen } from '@testing-library/react';
import { Avatar } from '@/shared/ui/Avatar';

describe('Avatar', () => {
  it('should render avatar', () => {
    render(<Avatar />);
    const avatar = screen.getByRole('img');
    expect(avatar).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<Avatar />);
    const avatar = screen.getByRole('img');
    expect(avatar).toMatchSnapshot();
  });
});
