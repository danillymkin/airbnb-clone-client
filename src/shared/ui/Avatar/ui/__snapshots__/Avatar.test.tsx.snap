// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<img
  alt="Avatar"
  class="rounded-full"
  data-nimg="1"
  decoding="async"
  height="30"
  loading="lazy"
  src="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75"
  srcset="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=32&q=75 1x, /_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75 2x"
  style="color: transparent;"
  width="30"
/>
`;
